client/my-sites/post-share/index.jsx

```diff
@@ -104,169 +141,313 @@ class PostShare extends Component {
 		);
 	}
 
-	dismiss = () => {
-		this.props.dismissShareConfirmation( this.props.siteId, this.props.post.ID );
-	};
+	renderShareButton() {
+		const {
+			hasRepublicizeSchedulingFeature,
+			siteId,
+			translate,
+		} = this.props;
+
+		const shareButton = <Button
+			className="post-share__button"
+			primary
+			onClick={ this.sharePost }
+			disabled={ this.isButtonDisabled() }
+		>
+			{ translate( 'Share post' ) }
+		</Button>;
+
+		if ( ! hasRepublicizeSchedulingFeature ) {
+			return shareButton;
+		}
 
-	sharePost = () => {
-		this.props.sharePost( this.props.siteId, this.props.post.ID, this.state.skipped, this.state.message );
-	};
+		return (
+			<div className="post-share__button-actions">
+				<Button onClick={ this.previewSharingPost }>
+					{ translate( 'Preview' ) }
+				</Button>
+
+				<ButtonGroup className="post-share__share-combo">
+					{ shareButton }
+
+					<CalendarButton
+						primary
+						className="post-share__schedule-button"
+						disabled={ this.isButtonDisabled() }
+						title={ translate( 'Set date and time' ) }
+						tabIndex={ 3 }
+						siteId={ siteId }
+						popoverPosition="bottom left" />
+				</ButtonGroup>
+			</div>
+		);
+	}
 
-	isButtonDisabled() {
-		if ( this.props.requesting ) {
-			return true;
+	renderUpgradeToGetPublicizeNudge() {
+		const { translate } = this.props;
+		return (
+			<Banner
+				className="post-share__upgrade-nudge"
+				feature="republicize"
+				title={ translate( 'Unlock the ability to re-share posts to social media' ) }
+				callToAction={ translate( 'Upgrade to Premium' ) }
+				description={ translate( 'Get unlimited premium themes, video uploads, monetize your site and more.' ) }
+			/>
+		);
+	}
+
+	renderUpgradeToGetSchedulingNudge() {
+		if ( this.props.hasRepublicizeSchedulingFeature ) {
+			return null;
 		}
 
-		return this.props.connections.filter( this.isConnectionActive ).length < 1;
+		const {
+			businessDiscountedRawPrice,
+			businessRawPrice,
+			translate,
+			userCurrency,
+		} = this.props;
+
+		return (
+			<Banner
+				className="post-share__footer-banner"
+				callToAction={
+					translate( 'Upgrade for %s', {
+						args: formatCurrency( businessDiscountedRawPrice || businessRawPrice, userCurrency ),
+						comment: '%s will be replaced by a formatted price, i.e $9.99'
+					} )
+				}
+				list={ [
+					translate( 'Schedule your social messages in advance.' ),
+					translate( 'Remove all advertising from your site.' ),
+					translate( 'Enjoy live chat support.' ),
+				] }
+				plan={ PLAN_BUSINESS }
+				title={ translate( 'Upgrade to a Business Plan!' ) } />
```

Comment:
Hi! I've found a possible matching string that has already been translated **19** times:
`translate( 'To upload themes, upgrade to Business Plan' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).